### PR TITLE
Update grype and syft

### PR DESCRIPTION
**Pull request by pbalogh-sa**

### CODE REVIEW

File: grype-server/pkg/scanner/scanner.go [hunk 1]
```diff
@@ -194,13 +208,20 @@ func (s *Scanner) loadBb(cfg *db.Config) error {
 	if status.Err != nil {
 		return fmt.Errorf("loaded DB has a failed status: %v", status.Err)
 	}
-	store, err := dbCurator.GetStore()
+	storeReader, dbCloser, err := dbCurator.GetStore()
+	if dbCloser != nil {
+		defer dbCloser.Close()
```
Comment (FrimIdan) [inline grype-server/pkg/scanner/scanner.go:213]: are we sure we want it? will it not delete the DB from filesystem?
not sure what it does
```
func (s *store) Close() {
	s.db.Exec("VACUUM;")

	sqlDB, err := s.db.DB()
	if err != nil {
		_ = sqlDB.Close()
	}
}
```

https://github.com/anchore/grype/blob/d5b825e40bbfc4696e7220c94c9f4469466e305e/grype/db/v4/store/store.go#L265

File: grype-server/pkg/scanner/scanner.go [hunk 2]
```diff
@@ -98,11 +100,16 @@ func (s *Scanner) ScanSbomJson(sbom string) (*models.Document, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to decode sbom: %v", err)
 	}
-	if formatOption == format.UnknownFormatOption {
-		return nil, fmt.Errorf("unknown SBOM format option: %v", formatOption)
+
+	// Validate returns error despite the syftSbom is valid so just log error and continue
```
Comment (FrimIdan) [inline grype-server/pkg/scanner/scanner.go:104]: not sure I understand, is it an error or not?

File: grype-server/pkg/scanner/scanner.go [hunk 3]
```diff
@@ -98,11 +100,16 @@ func (s *Scanner) ScanSbomJson(sbom string) (*models.Document, error) {
...
+	if err := formatOption.Validate(sbomReader); err != nil {
+		log.Errorf("unknown SBOM format option: %v, %v", formatOption.ID().String(), err)
+	}
+	if syftSbom.Artifacts.PackageCatalog == nil {
+		return nil, fmt.Errorf("packagecatalog is empty")
 	}
 
-	packages := grype_pkg.FromCatalog(syftSbom.Artifacts.PackageCatalog)
+	packages := grype_pkg.FromCatalog(syftSbom.Artifacts.PackageCatalog, grype_pkg.ProviderConfig{})
```
Comment (FrimIdan) [inline grype-server/pkg/scanner/scanner.go:112]: is it good to use an empty `grype_pkg.ProviderConfig{}`?

File: grype-server/pkg/scanner/scanner.go (hunk 1 above)
Comment (pbalogh-sa) [reply grype-server/pkg/scanner/scanner.go:213]: You are right it isn't necessary.
It closes the connections and frees the resources: https://github.com/anchore/grype/blob/f094b860b9b9ebcdf06d9b9c6f7d5c000fad395b/grype/db/db_closer.go#L5-L6
According to https://github.com/golang/go/blob/d11c58eedbee29b4912dd50508b36ad15ebb739e/src/database/sql/sql.go#L870-L876 it doesn't delete filesystem.

File: grype-server/pkg/scanner/scanner.go (hunk 3 above)
Comment (pbalogh-sa) [reply grype-server/pkg/scanner/scanner.go:112]: The only option that is used by `FromCatalog` is `GenerateMissingCPEs` that I forgot to set So I will set it true.
https://github.com/anchore/grype/blob/f094b860b9b9ebcdf06d9b9c6f7d5c000fad395b/grype/pkg/package.go#L66-L67

File: grype-server/pkg/scanner/scanner.go (hunk 2 above)
Comment (pbalogh-sa) [reply grype-server/pkg/scanner/scanner.go:104]: I got validation error: `unable to decode: EOF` from https://github.com/anchore/syft/blob/1cfa73732a4c2ce29117b55f0f7c0cff3133be3c/syft/formats/syftjson/validator.go#L22
The syftjson was created with syft. The validation shows this error. But the grype understand it. 
Maybe the validator needs to be fixed int syft.
https://mottaquikarim.github.io/dev/posts/is-json.decoder-broken-in-golang/

Comment (FrimIdan) [reply grype-server/pkg/scanner/scanner.go:104]: so why we need this validation?

Comment (FrimIdan) [reply grype-server/pkg/scanner/scanner.go:104]: It seems like something that was added by you, and I see that in the `Decode` function while calling to `IdentifyFormat(by)` the validation is being executed so we don't need to run it again.
https://github.com/anchore/syft/blob/1cfa73732a4c2ce29117b55f0f7c0cff3133be3c/syft/encode_decode.go#L29
https://github.com/anchore/syft/blob/1cfa73732a4c2ce29117b55f0f7c0cff3133be3c/syft/formats.go#L104

**Review by FrimIdan — APPROVED** [review]

**Pull request merged**